autorest/codegen/models/parameter_list.py

```diff
@@ -316,10 +320,105 @@ class GlobalParameterList(ParameterList):
     def implementation(self) -> str:
         return "Client"
 
+    @property
+    def code_model(self):
+        try:
+            return self._code_model
+        except AttributeError:
+            raise ValueError("You need to first set the code model")
+
+    @code_model.setter
+    def code_model(self, val):
+        self._code_model = val
+
+    @property
+    def endpoint_name(self) -> str:
+        return (
+            "endpoint" if self.code_model.options["version_tolerant"]
+            or self.code_model.options["low_level_client"]
+            else "base_url"
+        )
+
```

Comment:
**In [899](https://github.com/Azure/autorest.python/issues/899): In the case of no parameterized host, rename `base_url` to `endpoint`**

What is the meaning of `no parameterized host`? According to the code, rename happens when `version_tolerant` or `low_level_client`.

Reply:
sorry could've been worded more clearly. In swagger, we have `host` and [`x-ms-parameterized-host`](https://github.com/Azure/autorest/blob/main/docs/extensions/readme.md#x-ms-parameterized-host). By `parameterized host`, I mean if users used `x-ms-parameterized-host`. So in the case of `vesrion_tolerant` or `low_level_client`, and the users didn't use `x-ms-parameterized-host`, change the name of `base_url` to `endpoint`